source/NVDAObjects/UIA/__init__.py

```diff
@@ -1350,6 +1350,20 @@ def _get_children(self):
 			children.append(self.correctAPIForRelation(UIA(windowHandle=windowHandle,UIAElement=e)))
 		return children
 
+	def getChild(self, index):
+		childrenCacheRequest = UIAHandler.handler.baseCacheRequest.clone()
+		childrenCacheRequest.TreeScope = UIAHandler.TreeScope_Children
+		try:
+			cachedChildren = self.UIAElement.buildUpdatedCache(childrenCacheRequest).getCachedChildren()
+		except COMError:
+			return super(UIA, self).children[index]
+		if not cachedChildren:
+			# GetCachedChildren returns null if there are no children.
+			raise IndexError

```

Comment:
This is confusing. some instances of getChild return None when out of range, the base implementation raises an index error. I think the former should be preferred, as we aren't indexing a list here.